IM/connectors/OCCI.py

```diff
@@ -1271,6 +1275,57 @@ def get_keystone_token(occi, keystone_uri, auth):
         """
         Contact the specified keystone server to return the token
         """
+        version = KeyStoneAuth.get_keystone_version(occi, keystone_uri, auth)
+        if version == 2:
+            occi.logger.debug("Getting Keystone v2 token")
+            return KeyStoneAuth.get_keystone_token_v2(occi, keystone_uri, auth)
+        elif version == 3:
+            occi.logger.debug("Getting Keystone v3 token")
+            return KeyStoneAuth.get_keystone_token_v3(occi, keystone_uri, auth)
+        else:
+            # this must never happen
+            raise Exception("Error obtaining Keystone Token: Unknown version %d" % version)
+
+    @staticmethod
+    def get_keystone_version(occi, keystone_uri, auth):
+        """
+        Contact the specified keystone server to return version to use
+        """
+        version = None
+        token = auth and "token" in auth
+
+        try:
+            uri = uriparse(keystone_uri)
+            server = uri[1].split(":")[0]
+            port = int(uri[1].split(":")[1])
+
+            url = "https://%s:%s" % (server, port)
```

Comment:
Extracting `server` and `port` from Keystone URI will break compatibility as many setups have path-based service separation. For example, you could receive `https://my-ostack.example.org/identity` as Keystone URI and you would be expected to just append `/v3/...` and make requests. Is there another reason for doing this? @enolfc ?

Reply:
+1 @arax, just use the received URI. Moreover you don't seem to use it later on for anything, just for recomposing the URI